services/QuillLMS/app/queries/quill_big_query/concerns/view_query_fallback.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module QuillBigQuery
+  module Concerns
+    module ViewQueryFallback
+      extend ActiveSupport::Concern
+
+      class BrokenViewError < StandardError; end
+
+      BROKEN_VIEW_ERRORS = [
+        ::Google::Cloud::NotFoundError,
+        ::Google::Cloud::InvalidArgumentError
+      ]
+
+      def fallback_views = raise NotImplementedError
+      def run_query = run_query_with_fallback
+
+      def run_query_with_fallback
+        query_runner(query)
+      rescue *BROKEN_VIEW_ERRORS => e
+        ErrorNotifier.report(BrokenViewError.new(e.message))
```

Comment:
Is there any more context that would be useful to include here,  such as view key or query?  (It might be in e.message but I can't really tell).

Reply:
The message logs the full query. There might be additional info we could like like query class. Let me think on that. Long term, I'm debating moving away from this query fallback anyway if we fully move off of MaterializedViews (we'd adjust the tables to only refresh once a day overnight, so there'd only be about 30 seconds in the middle of the night where the query might error.